libdistopia/libdistopia/lib/distance_array_dispatch.h

```diff
@@ -0,0 +1,33 @@
+#ifndef DISTOPIA_DISTANCE_ARRAY_DISPATCH_H
+#define DISTOPIA_DISTANCE_ARRAY_DISPATCH_H
+
+#include <cstddef>
+#include <cstdio>
+#include "distopia_type_traits.h"
+
+template <typename T>
+void DistanceArrayOrthoDispatch(const T *coords0, const T *coords1, const T *box,
+                                std::size_t n0, std::size_t n1, T *out);
+
+using DistanceArrayOrtho_FptrT = decltype(&DistanceArrayOrthoDispatch<float>);
+using DistanceArrayOrtho_DptrT = decltype(&DistanceArrayOrthoDispatch<double>);
+
+enum selectT {Float = 0, Double = 1};
```

Comment:
@richardjgowers This stuff is reusable from the other header, I will make it a  seperate header in another PR.

Reply:
Actually cancel that see my other comment.